### DISCUSSION TITLE
Error: Body exceeded 10mb limit

### DISCUSSION THREAD

**Discussion by iiome** [vercel/vercel#4615] [Help] [answered]
Hi

When deploying using vercel command line I face an error `Error: Body exceeded 10mb limit`.

The command `vercel -d` would hint anything about which file(s) create the problem.

This is a next.js project. So I run `next build` locally. I can only find one file that is larger than 8mb in the project.

```
$ find . -size +8000kb
./.next/static/development/pages/_app.js
$ ls -lh ./.next/static/development/pages/_app.js
-rw-r--r--  1 me  staff    26M 10 Jun 08:19 .next/static/development/pages/_app.js
```
The project features more that 35,000 SGG paths (HTML + JSON).

Any help, comment, idea on how to fix this is appreciated.

Please find below the relevant log lines.

```
11:46:48.612  + First Load JS shared by all     58 kB
11:46:48.612    ├ static/pages/_app.js          287 B
11:46:48.612    ├ chunks/commons.94e943.js      10.7 kB
11:46:48.612    ├ chunks/framework.e84fa6.js    40 kB
11:46:48.612    ├ runtime/main.7965a5.js        6.28 kB
11:46:48.612    ├ runtime/webpack.c21266.js     746 B
11:46:48.612    └ css/8dc9ec173b84701f1627.css  4.19 kB
11:46:48.612  λ  (Lambda)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
11:46:48.612  ○  (Static)  automatically rendered as static HTML (uses no initial props)
11:46:48.612  ●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
11:46:48.681 Done in 52.59s.
11:47:01.452 Traced Next.js serverless functions for external files in: 6258.320ms
11:47:02.627 Compressed shared serverless function files: 1175.017ms
11:47:03.143 All serverless functions created in: 514.692ms
11:55:52.025 Error: Body exceeded 10mb limit
11:55:54.078 Done with "package.json"
```

**Comment by styfle** [comment, marked as answer]
Hi @iiome 

Your output of 35,000 files exceeds the maximum 16,000. https://vercel.com/docs/v2/platform/limits#files

We are working to increase this limit and we'll also make the error message a little better, thanks!

**Reply by meronek** [reply]
Thanks for the clarification there. I am working on a site with a ton of pre-built files/paths and hitting limits. The workaround I'm using right now is generating only a few paths and leaving the rest to be built at runtime with router.isFallback. 

Question: if my isFallback thing ends up generating more than 16,000 files, will I run into trouble?

**Reply by styfle** [reply]
No, using Next.js `isFallback` to dynamically generate pages at runtime does not count towards this file limit. This property creates a response on demand that gets cached, but there is no file generated.

**Comment by iiome** [comment]
Hi @styfle 

Thanks for quick reply and the help!
I did check the limits but I was mislead by the mention of file size.

**Comment by nickluger** [comment]
Faced the same issue. File size error message is cryptic/misleading.
